CONTRIBUTING.md

```diff
@@ -1,51 +1,193 @@
+    repositories to keep track of: the original ManimCommunity repo, your own
+    fork of it, and your local repository.
 
-1. First, make a fork of this repository.
-2. Then clone the repository you forked using the command below (bash/zsh).
+3. You have to make `git` be aware of the three repositories, by entering the
+   commands below.
 
- ```sh
-   git clone <your-fork-url>
-   cd manim
- ```
+   ```sh
+   git remote add upstream https://github.com/ManimCommunity/manim.git
+   git pull upstream master
+   ```
 
-3. Now that you have cloned the repo, before making any changes, you have to enter the commands below in order to be able to contribute to Manim Community.
+	After these commands, your local repository can keep track of your fork
+    (referred to as 'origin') as well as the main ManimCommunity repository
+    (referred to as 'upstream').
 
+4. Choose the correct branch for your changes.  If you are working on a branch
+   that already exists on the main repository, you can execute
+   ```sh
+   git checkout -b <branch-name> upstream/master
+   ```
+   
+   If you are starting a new branch, execute
    ```sh
-   git remote add fork https://github.com/ManimCommunity/manim.git
-   git fetch fork
-   git checkout -b <your-branch-name> fork/master
+   git checkout -b <new_branch-name>
    ```
 
-   The first command tells `git` that you are going to make a Pull Request to Manim Community. 
-   The second command pulls all the commits from the aforementioned fork.
-   Finally, the third one makes your current working branch up-to-date with Manim Community's master branch.
+	You can now make any changes to your local repository, add them, and commit
+   them.
 
-    Now there are three repositories that git is keeping track of: the manim community repo (referred to as "fork"), your own fork of it (referred to by git as "origin"), and your local repository.
+5. To make your changes visible in your GitHub fork, instead of typing in `git
+   push` as usual, you need to enter the command below.
 
-4. After that, you can make your changes to the repo's files (the code is in the `manim` directory). Then, you can commit said changes.
+   ```sh
+   git push -u origin <branch-name>
+   ```
 
-> As stated in the General Contribution Guidelines, make sure to test your changes against the example scenes in the `example_scenes` directory and test scenes in the `tests` directory, and add a new test in the `tests` directory for your new feature if you've made one.
+   Doing so creates a new branch with the updated contents of your fork on
+   GitHub (the 'origin').
 
-5. Finally, instead of  typing in `git push`, enter the command below.
+6. If you want your changes to be incorporated to the main ManimCommunity
+   repository, you need to make sure that there are no merge conflicts between
+   the current upstream/master and the changes you are trying to make.  For
+   this purpose, please execute `git pull origin master` before you push
```

Comment:
Isn't it `git pull upstream master`?
```suggestion
   this purpose, please execute `git pull upstream master` before you push
```
This statement mentioned in original code will just pull the master branch of their fork which is unnecessary, it should pull from master branch on manim community.